codegen/src/serialization.rs

```diff
@@ -0,0 +1,87 @@
+use std::collections::BTreeMap;
```

Comment:
Please.  It's a bit of an edge case we don't run into with other uses of Serde in the project so documenting it will help us remember why it's there.